### PR TITLE
feat: Update to new version of Web3Modal

### PR DESCRIPTION
**Pull request by efstajas**
Replaces Web3Modal with new version, using WalletConnect Protocol V2. Also updates to TypeScript V5.

### CODE REVIEW

File: src/lib/stores/wallet/wallet.store.ts [hunk 1]
```diff
@@ -1,39 +1,42 @@
...
-const WEB_3_MODAL_PROVIDER_OPTIONS = {
-  walletconnect: {
-    package: WalletConnectProvider,
-    options: {
-      infuraId: INFURA_ID,
-    },
-  },
-};
+const WAGMI_CHAINS = [mainnet, goerli];
+const WALLETCONNECT_PROJECT_ID = 'c09f5d8545d67c604ccf454219fd8f4d';
```
Comment (jtourkos) [inline src/lib/stores/wallet/wallet.store.ts:39]: Should we make this "private"?

**Review by jtourkos — APPROVED** [review]

File: src/lib/stores/wallet/wallet.store.ts (hunk 1 above)
Comment (efstajas) [reply src/lib/stores/wallet/wallet.store.ts:39]: It's not secret, so I think we can keep it in here.

Comment (jtourkos) [reply src/lib/stores/wallet/wallet.store.ts:39]: (Didn't notice before approving but svelte-check's failing)

Comment (efstajas) [reply src/lib/stores/wallet/wallet.store.ts:39]: yep i saw. should be resolved now

**Comment by efstajas** [comment]
Closing this since we decided to go with `web3-onboard`.

**Pull request closed without merging**